src/main/java/org/broadinstitute/hellbender/engine/SlidingWindowWalker.java

```diff
@@ -0,0 +1,116 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMSequenceRecord;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import java.util.ArrayList;
+
+/**
+ * An SlidingWindowWalker is a tool that processes a single window over the genome (constructed with a window-size and window-step)
+ * at a time, with the ability to query optional overlapping sources of reads, reference data, and/or variants/features.
+ *
+ * SlidingWindow authors must implement the apply() method to process each position, and may optionally implement
+ * onTraversalStart() and/or onTraversalDone().
+ *
+ * @author Daniel Gómez-Sánchez (magicDGS)
+ */
+public abstract class SlidingWindowWalker extends GATKTool {
+
+	@Argument(fullName = "windowSize", shortName = "windowSize", doc = "Window size for iterate over the variants", common = false, optional = false)
+	public int windowSize;
+
+	@Argument(fullName = "windowStep", shortName = "windowStep", doc = "Window step for iterate over the variants", common = false, optional = false)
+	public int windowStep;
+
+	/**
+	 * This is the array of intervals that configure the windows included in the tool
+	 */
+	protected final ArrayList<SimpleInterval> windows = new ArrayList<>();
+
+	/**
+	 * Initialize data sources for traversal.
+	 *
+	 * Marked final so that tool authors don't override it. Tool authors should override onTraversalStart() instead.
+	 */
+	@Override
+	protected final void onStartup() {
+		super.onStartup();
+		// the tool needs a sequence dictionary for get the windows
+		if(getBestAvailableSequenceDictionary() == null) {
+			throw new UserException("Tool " + getClass().getSimpleName() + " requires some source for sequence dictionary, but none were provided");
+		}
+		// construct the windows for the available sequence dictionary
+		constructWindows();
+	}
+
+	/**
+	 * Construct the windows to iterate from.
+	 */
+	private void constructWindows() {
+		logger.info("Computing windows for the sequence");
+		for(final SAMSequenceRecord seq: getBestAvailableSequenceDictionary().getSequences()) {
+			for(int i = 1; i <= seq.getSequenceLength(); i+=windowStep) {
+				final SimpleInterval currentWindow = new SimpleInterval(seq.getSequenceName(), i, i+windowSize);
```

Comment:
Should you trim the final window in each contig at the contig length to avoid going out of bounds? There is an `IntervalUtils.trimIntervalToContig()` method that you can use for this.